learning/courses/quantum-machine-learning/data-encoding.ipynb

```diff
@@ -0,0 +1,1713 @@
+    "$$\n",
+    "\\mathscr{U}^{(r)}_{\\text{ZFM}}\\left(\\vec{x}\\right)=\\prod_{s=1}^{r}\\left[\\left(\\bigotimes_{k = 1}^N P(\\vec{x}_k)\\right)H^{\\otimes N}\\right]\n",
+    "$$\n",
+    "The data features, $x_k$, are mapped to the phase gates in the same way in all $r$ repetitions. The ZFM feature map state is a product state and is efficient for classical simulation.<sup>4</sup>\n",
+    "\n",
+    "To start with a small example, a 2-qubit ZFM circuit is coded using Qiskit and drawn to display the simple circuit structure. In the example, a single repetition, $r=1$, is implemented with the data vector $\\vec{x} = \\left(\\textstyle\\frac{1}{2}\\pi, \\textstyle\\frac{1}{3}\\pi\\right)$. The ZFM circuit unitary operator acts on the initial state in the following way:\n",
+    "\n",
+    "$$\n",
+    "\\mathscr{U}_{\\text{ZFM}}(\\bar{x})|00\\rangle = P(\\bar{x})^{\\otimes 2} H^{\\otimes 2}|00\\rangle = \\left( P\\left(\\textstyle\\frac{1}{2}\\pi\\right)H|0\\rangle \\right) \\otimes \\left(P\\left(\\textstyle\\frac{1}{3}\\pi\\right)H|0\\rangle\\right).\n",
+    "$$\n",
+    "\n",
+    "The formula has been rearranged around the tensor product to emphasize the operations on each qubit. The following Qiskit code uses Hadamard and phase gates explicitly to show the structure of the ZFM:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 22,
+   "id": "f5c70df4-faea-4817-a870-95638eb97dbd",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<Image src=\"/learning/images/courses/quantum-machine-learning/data-encoding/extracted-outputs/f5c70df4-faea-4817-a870-95638eb97dbd-0.avif\" alt=\"Output of the previous code cell\" />"
+      ]
+     },
+     "execution_count": 22,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "qc0 = QuantumCircuit(1)\n",
+    "qc1 = QuantumCircuit(1)\n",
+    "\n",
+    "qc0.h(0)\n",
+    "qc0.p(pi / 2, 0)\n",
+    "\n",
+    "qc1.h(0)\n",
+    "qc1.p(pi / 3, 0)\n",
+    "\n",
+    "# Combine circuits qc0 and qc1 into 1 circuit\n",
+    "qc = QuantumCircuit(2)\n",
+    "qc.compose(qc0, [0], inplace=True)\n",
+    "qc.compose(qc1, [1], inplace=True)\n",
+    "\n",
+    "qc.draw(\"mpl\", scale=1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b1be7f51-3a4e-4d2e-8d51-d3ac62f4fd88",
+   "metadata": {},
+   "source": [
+    "We now encode the same data vector $\\vec{x} = \\left(\\textstyle\\frac{1}{2}\\pi, \\textstyle\\frac{1}{3}\\pi\\right)$ to a ZFM circuit with three repetitions, $r=3$, using the Qiskit [`ZFeatureMap`](https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.library.ZFeatureMap) class, which altogether gives us the quantum feature map $\\mathscr{U}_{\\text{ZFM}}(\\vec{x})$. By default in the `ZFeatureMap` class, parameters $\\beta$ are multiplied by 2 before mapping to the phase gate $\\beta \\rightarrow P(\\theta = 2\\beta)$. To reproduce the same encodings as above, we divide by 2."
```

Comment:
Can you please find and replace `https://docs.quantum.ibm.com/` for `/docs/`, e.g. `/docs/api/qiskit/qiskit.circuit.library.ZFeatureMap`